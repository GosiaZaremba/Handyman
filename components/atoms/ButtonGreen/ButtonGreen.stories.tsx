import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {ButtonGreen} from './ButtonGreen';

storiesOf('atoms/ButtonGreen', module)
  .addDecorator(story => story())
  .add('default', () => (
    <ButtonGreen buttonText={'Button Text'} onPress={() => {}} />
  ));
