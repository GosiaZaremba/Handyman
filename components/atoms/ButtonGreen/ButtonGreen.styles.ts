import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  gradient: {
    borderRadius: 10,
    width: '100%',
    height: 35,
    justifyContent: 'center',
    alignItems: 'center',
    elevation: 8,
  },
  pressable: {
    borderRadius: 10,
    width: '100%',
    height: 35,
    justifyContent: 'center',
    alignItems: 'center',
  },
  pressed: {
    opacity: 0.8,
    borderRadius: 10,
    width: '100%',
    height: 35,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    color: '#fff',
    fontFamily: 'OpenSans-Bold',
  },
});
