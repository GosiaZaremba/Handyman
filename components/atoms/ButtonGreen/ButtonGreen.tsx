import React from 'react';
import {Pressable, Text} from 'react-native';
import styles from './ButtonGreen.styles';
import LinearGradient from 'react-native-linear-gradient';
import {Colors} from '../../../constants/colors';

export type Props = {
  buttonText: string;
  onPress: () => void;
  testID?: string;
  testOnly_pressed?: boolean;
};

export const ButtonGreen: React.FC<Props> = ({
  buttonText,
  onPress,
  testID,
  testOnly_pressed,
}) => {
  return (
    <LinearGradient
      colors={[Colors.button.light, Colors.button.dark]}
      start={{x: 0, y: 0.5}}
      end={{x: 1, y: 0.5}}
      style={styles.gradient}>
      <Pressable
        style={({pressed}) => (pressed ? styles.pressed : styles.pressable)}
        onPress={onPress}
        testID={testID}
        testOnly_pressed={testOnly_pressed}>
        <Text style={styles.buttonText}>{buttonText}</Text>
      </Pressable>
    </LinearGradient>
  );
};
