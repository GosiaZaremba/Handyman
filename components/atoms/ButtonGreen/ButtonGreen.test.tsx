import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {ButtonGreen} from './ButtonGreen';

const mockOnPress = jest.fn();

describe('Atom AddPhotoButton', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <ButtonGreen
          onPress={mockOnPress}
          buttonText={'Button Green'}
          testID="atom-buttonGreen"
          testOnly_pressed={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure text button is passed correctly', () => {
    const {getByTestId} = render(
      <ButtonGreen
        onPress={mockOnPress}
        buttonText={'Button Green'}
        testID="atom-buttonGreen"
        testOnly_pressed={false}
      />,
    );
    const addPhotoButton = getByTestId('atom-buttonGreen');
    expect(addPhotoButton).toHaveTextContent('Button Green');
  });

  test('ensure onPress works correctly', () => {
    const {getByTestId} = render(
      <ButtonGreen
        onPress={mockOnPress}
        buttonText={'Button Green'}
        testID="atom-buttonGreen"
        testOnly_pressed={false}
      />,
    );
    const addPhotoButton = getByTestId('atom-buttonGreen');
    fireEvent.press(addPhotoButton);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(addPhotoButton.props.style.borderRadius).toBe(10);
    expect(addPhotoButton.props.style.opacity).toBe(undefined);
  });

  test('ensure pressable pressed works correctly', () => {
    const {getByTestId} = render(
      <ButtonGreen
        onPress={mockOnPress}
        buttonText={'Button Green'}
        testID="atom-buttonGreen"
        testOnly_pressed={true}
      />,
    );
    const buttonGreen = getByTestId('atom-buttonGreen');
    fireEvent.press(buttonGreen);
    expect(buttonGreen.props.style.borderRadius).toBe(10);
    expect(buttonGreen.props.style.opacity).toBe(0.8);
  });
});
