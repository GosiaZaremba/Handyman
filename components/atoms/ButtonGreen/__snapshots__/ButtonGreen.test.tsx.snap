// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom AddPhotoButton Component Snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278363496,
      4285643865,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0.5,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0.5,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 10,
      "elevation": 8,
      "height": 35,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "height": 35,
        "justifyContent": "center",
        "opacity": 0.8,
        "width": "100%",
      }
    }
    testID="atom-buttonGreen"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "OpenSans-Bold",
        }
      }
    >
      Button Green
    </Text>
  </View>
</BVLinearGradient>
`;
